clients/src/main/java/org/apache/kafka/common/network/SSLTransportLayer.java

```diff
@@ -222,11 +222,13 @@ public void handshake() throws IOException {
                               channelId, appReadBuffer.position(), netReadBuffer.position(), netWriteBuffer.position());
                     handshakeResult = handshakeWrap(write);
                     if (handshakeResult.getStatus() == Status.BUFFER_OVERFLOW) {
-                        int currentPacketBufferSize = packetBufferSize();
-                        netWriteBuffer = Utils.ensureCapacity(netWriteBuffer, currentPacketBufferSize);
-                        if (netWriteBuffer.position() >= currentPacketBufferSize) {
+                        int currentNetWriteBufferSize = netWriteBufferSize();
+                        netWriteBuffer.position(netWriteBuffer.limit());
```

Comment:
Yes, you are right. Should do `compact()` before `Utils.ensureCapacity()` instead of just setting `position`. Will fix that.